Tests/UnitTests/Plugins/GeoModel/GeoModelMuonMockupBluePrintTests.cpp

```diff
@@ -0,0 +1,236 @@
+using BoundingBoxesVols = std::vector<
+    std::pair<Acts::GeoModelDetectorObjectFactory::GeoModelBoundingBox,
+              Acts::GeoModelDetectorObjectFactory::GeoModelVolumeBox>>;
+
+// function that builds the chamber tracking volumes
+std::shared_ptr<StaticBlueprintNode> buildBarrelNode(
+    const BoundingBoxesVols& boundingBoxes,
+    const SensitiveSurfaces& sensitiveSurfaces, const std::string& name,
+    const Logger& logger) {
+  // the vector with the chamber nodes and the volumes
+  std::vector<std::shared_ptr<StaticBlueprintNode>> volChamberNodes;
+  std::vector<std::unique_ptr<Acts::TrackingVolume>> volChambers;
+
+  for (std::size_t sector = 0; sector < nSectors; sector++) {
+    ACTS_DEBUG("Barrel name: " << name << " sector: " << sector);
+    // Find the bounding box for the given chamber name and sector
+    auto it_first =
+        std::ranges::find_if(boundingBoxes, [&name, &sector](const auto& pair) {
+          return pair.first->name().find(name) != std::string::npos &&
+                 pair.first->name().find("_1_" + std::to_string(sector + 1)) !=
+                     std::string::npos;
+        });
+
+    // Check if the bounding box was found
+    BOOST_CHECK(it_first != boundingBoxes.end());
+
+    auto it_second = std::find_if(
+        it_first, boundingBoxes.end(), [&name, &sector](const auto& pair) {
+          return pair.first->name().find(name) != std::string::npos &&
+                 pair.first->name().find("_1_" + std::to_string(sector + 1)) !=
+                     std::string::npos;
+        });
+    // Check if the second bounding box was found
+    BOOST_CHECK(it_second != boundingBoxes.end());
+
+    auto vol1 = it_first->second;
+    auto vol2 = it_second->second;
+    Vector3 center = (vol1->center() + vol2->center()) / 2;
+    Acts::Transform3 volTrf = vol1->transform().rotation() *
+                              Acts::Transform3(Acts::Translation3(center));
+    auto volumeBounds = vol1->volumeBounds().values();
+    double ymin = std::min(vol1->center().y(), vol2->center().y());
+    double ymax = std::max(vol1->center().y(), vol2->center().y());
+
+    double yup = ymax + volumeBounds[1];
+    double ylow = ymin - volumeBounds[1];
+
+    std::shared_ptr<Acts::CuboidVolumeBounds> chamberBounds =
+        std::make_shared<Acts::CuboidVolumeBounds>(
+            volumeBounds[0], (yup - ylow), volumeBounds[2]);
+
+    std::unique_ptr<Acts::TrackingVolume> chamberVolume =
+        std::make_unique<Acts::TrackingVolume>(
+            volTrf, chamberBounds,
+            "Chamber_" + name + "_" + std::to_string(sector + 1));
+
+    // add the multilayer volumes
+    chamberVolume->addVolume(std::make_unique<TrackingVolume>(
+        *vol1, "Chamber_" + name + "_" + std::to_string(sector + 1) + "_MDT1"));
+    chamberVolume->addVolume(std::make_unique<TrackingVolume>(
+        *vol2, "Chamber_" + name + "_" + std::to_string(sector + 1) + "_MDT2"));
```

Comment:
What you're doing here is essentially fully assembling the cuboids which contain the individual multilayers, correct?

This is one way to do it, especially given the fact that there's currently no way to make a node that produces an *auto-sized* wrapping volume. I think down the road, I'll want to change this to have a "wrapper cuboid" node type that produces a cuboid that self-sizes to fit around the children that it receives.

That way, it would not be necessary to calculate the correct sizing for the chamber volume individually here.

What do you think @dimitra97?